src/components/TodoItem.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class TodoItem extends React.Component {
+  state = {
+    editTitle: this.props.todo.title,
+  }
+
+  editCurrentTitle = (e) => {
+    this.setState({
+      editTitle: e.target.value,
+    });
+  }
+
+  render() {
+    const { todo,
+      statusOfTodo,
+      handleRemoveTodo,
+      handleEditTodo,
+      handleEditTitle } = this.props;
+    const { id, title, completed } = todo;
+    const { editTitle } = this.state;
+
+    return (
+      <>
+        <div className="view">
+          <input
+            type="checkbox"
+            className="toggle"
+            id={id}
+            checked={completed}
+            onChange={() => statusOfTodo(id)}
+          />
+          <label onDoubleClick={() => handleEditTodo(id, title)}>{title}</label>
```

Comment:
onDoubleClick you have a method that had to enable editing mode and write title from props into title in this.state from this component, method for this event handler should be created in this component;
all submitting stuff should be done on input event;

this way you will solve the problem